crates/stages/src/stages/execution.rs

```diff
@@ -336,72 +346,84 @@ impl<EF: ExecutorFactory, DB: Database> Stage<DB> for ExecutionStage<EF> {
+            // get all batches for account change
+            // Check if walk and walk_dup would do the same thing
+            let account_changeset_batch =
+                account_changeset.walk_range(range.clone())?.collect::<Result<Vec<_>, _>>()?;
+
+            // revert all changes to PlainState
+            for (_, changeset) in account_changeset_batch.into_iter().rev() {
+                if let Some(account_info) = changeset.info {
+                    tx.put::<tables::PlainAccountState>(changeset.address, account_info)?;
+                } else {
+                    tx.delete::<tables::PlainAccountState>(changeset.address, None)?;
+                }
             }
-        }
 
-        // get all batches for storage change
-        let storage_changeset_batch = storage_changeset
-            .walk_range(BlockNumberAddress::range(range.clone()))?
-            .collect::<Result<Vec<_>, _>>()?;
+            // get all batches for storage change
+            let storage_changeset_batch = storage_changeset
+                .walk_range(BlockNumberAddress::range(range.clone()))?
+                .collect::<Result<Vec<_>, _>>()?;
 
-        // revert all changes to PlainStorage
-        let mut plain_storage_cursor = tx.cursor_dup_write::<tables::PlainStorageState>()?;
+            // revert all changes to PlainStorage
+            let mut plain_storage_cursor = tx.cursor_dup_write::<tables::PlainStorageState>()?;
 
-        for (key, storage) in storage_changeset_batch.into_iter().rev() {
-            let address = key.address();
-            if let Some(v) = plain_storage_cursor.seek_by_key_subkey(address, storage.key)? {
-                if v.key == storage.key {
-                    plain_storage_cursor.delete_current()?;
+            for (key, storage) in storage_changeset_batch.into_iter().rev() {
+                let address = key.address();
+                if let Some(v) = plain_storage_cursor.seek_by_key_subkey(address, storage.key)? {
+                    if v.key == storage.key {
+                        plain_storage_cursor.delete_current()?;
+                    }
+                }
+                if storage.value != U256::ZERO {
+                    plain_storage_cursor.upsert(address, storage)?;
                 }
             }
-            if storage.value != U256::ZERO {
-                plain_storage_cursor.upsert(address, storage)?;
-            }
-        }
 
-        // Discard unwinded changesets
-        tx.unwind_table_by_num::<tables::AccountChangeSet>(unwind_to)?;
+            // Discard unwinded changesets
+            let mut rev_acc_changeset_walker = account_changeset.walk_back(None)?;
+            while let Some((block_num, _)) = rev_acc_changeset_walker.next().transpose()? {
+                if block_num <= unwind_to {
+                    break
+                }
+                // delete all changesets
+                rev_acc_changeset_walker.delete_current()?;
+            }
```

Comment:
sneaked this one unknowingly during the merge commit. But it does seem correct, right? It was being unwinded twice right?